apis/deployer/mock/types_provider.go

```diff
@@ -22,6 +22,10 @@ type ProviderConfiguration struct {
 	// Phase sets the phase of the DeployItem
 	Phase *lsv1alpha1.ExecutionPhase `json:"phase,omitempty"`
 
+	// InitialPhase sets the phase of the DeployItem, but only if it is empty or "Init"
+	// Additionally, setting it will suppress the DeployItem phase being set to "Succeeded" after successful reconciliation
+	InitialPhase *lsv1alpha1.ExecutionPhase `json:"initialPhase,omitempty"`
```

Comment:
why is the default phase not enough?

Reply:
Because the deployer sets the phase in each reconcile loop. This means I can't test whether the phase is `Failed` after it has been in `Progressing` for too long, because it will immediately become `Progressing` again. Therefore I need an option to set the phase to `Progressing`, but don't overwrite the `Failed` afterwards.